src/Traits/DetectsChanges.php

```diff
@@ -18,7 +18,8 @@ protected static function bootDetectsChanges()
 
                 //temporary hold the original attributes on the model
                 //as we'll need these in the updating event
-                $oldValues = (new static)->setRawAttributes($model->getOriginal());
+                $originalAttributes = $model->getOriginal() + $model->fresh()->attributesToArray();
```

Comment:
I'm not happy with this line. It's an additional DB query. And isn't it only required for `creating`? After it's created it doesn't need to check for any DB defaults!?
@freekmurze should we add a new optional static model property `$hasDefaultDbAttributes` (name only as idea) to prevent this DB query for users without DB defaults or using the `$attributes` approach?

Reply:
@Gummibeer That's a good thought. 

I was looking for a way to limit the extra DB call to only those cases where it was needed, i.e. where the model has default values. I tried to determine if the model had default values set in its migration. However, I did not find any clean ways of achieving this.